netmiko/flexvnf/flexvnf_ssh.py

```diff
@@ -41,27 +42,29 @@ def enter_cli_mode(self):
                 break
             count += 1
 
-    def check_enable_mode(self, *args, **kwargs):
+    def check_enable_mode(self, *args, **kwargs) -> bool:
         """No enable mode on flexvnf."""
-        pass
+        return False
 
-    def enable(self, *args, **kwargs):
+    def enable(self, *args, **kwargs) -> str:
         """No enable mode on flexvnf."""
-        pass
+        return ""
 
-    def exit_enable_mode(self, *args, **kwargs):
+    def exit_enable_mode(self, *args, **kwargs) -> str:
         """No enable mode on flexvnf."""
-        pass
+        return ""
 
-    def check_config_mode(self, check_string="]"):
+    def check_config_mode(self, check_string: str = "]", *args, **kwargs) -> bool:
```

Comment:
I don't like relying on `*args, **kwargs`.  🤷 

since original signature doesn't match base class (triggering mypy error), existing code would outright break if users tried to follow signature from base class.  This silently swallows extra parameters.  Alternatively we could pass the extras (`pattern` in this case) on to the Super.  Not sure which makes more sense.

Reply:
@wrgeorge1983 The difficult on the other side is `if I ever add an argument to the base class, then I have 50+ places I  also need to update it in the child classes`.

I am fine with either pattern though (i.e. either the below) or fully defining all of the arguments.

`def exit_enable_mode(self, *args: Any, **kwargs: Any) -> str:`

I am pretty sure the above will pass mypy.

Note, if people are passing any extra arguments here there code really is broken and we are doing a major release (i.e. netmiko3 to netmiko4) so they should be testing the new code before they upgrade.

Reply:
Or in the case of `check_config_mode`, the form should be one of the following two:

```
def check_config_mode(self, check_string: str = "]", *args: Any, **kwargs: Any) -> bool:
```

Or

```
def check_config_mode(self, check_string: str = "*>", pattern: str = "") -> bool:
```